doc/decisions/error_message.md

```diff
@@ -0,0 +1,215 @@
+```
+
+The error #<NR> will be the color red while <PLUGIN> will be the color blue.
+
+Optionally a third line indicating a solution can be added. Eg. for a permission related error there would be a third line:
+
+```
+Possible Solution: Retry the command as sudo (sudo !!)
+```
+
+To avoid losing information, the user can use the command line argument `-v` (verbose) to show
+`Mountpoint`, `Configfile` in addition to the current error message.
+Furthermore a developer can use the command line argument `-d` (debug)
+to show `At` for debugging purposes.
+
+All "fatal" errors will be converted to "errors" as the distinguishment is not relevant.
+
+Unused marked errors will be removed from the specification.
+
+Errors will be categorizes into logical groups with subgroups.
+Each error will be made up of 5 characters, where the first 2 character indicate the highest level
+and character 3 to 5 will be used for subgrouping.
+
+- Permanent errors (01000)
+  - Resource (01100)
+  - Parsing (01200)
+  - Installation (01300)
+  - Logical (01400)
+- Conflict (02000)
+- Timeout (03000)
+- Validation (04000)
+  - Syntactic (04100)
+  - Semantic (04200)
+
+## Rationale
+
+The new error message is much more succinct which gives end users more relevant information.
+Furthermore the solution approach still holds all necessary information if requested by users.
+
+A new metakey `error/solution` will be implemented and attached to the parentkey via the new API.
+`Ingroup`, `Description` and `Module` will be removed from the error message as they provide no useful
+information at all.
+
+The grouping of errors will allow developers to filter for specific as well as more general errors to correctly
+react to them programmatically.
+The new concept will permit additional subgrouping of errors in case it might be needed in the future. Imagine the case where
+"Resource" errors is too general because developers saw a need for splitting the errors in "permission" and "existence" errors.
+They can simply take the current subclass 01100 and make "permission" be 01110 and "existence" be 01120. This will also allow
+backwards compatibility by applications just checking for resource errors in general.
+Splitting/merging/rearranging any category should only be done by a decision (such as this file here) because elektra developers
+should not be able to generate a new category as they wish because it would lead to the same proliferation of errors as we have now.
+
+The classification as of now is done with the following concept in mind:
+
+- Permanent errors
+    Generally errors which require changes by a user which are not getting fixed by eg. retrying.
+- Resource
+    The class of errors which require certain permissions (opening a file) or where some directory/file is missing.
+    One possible reaction to these kind of errors is to try a different directory/file or create a missing resource.
+- Parsing
+    Errors while parsing files such as ini/yml/etc.
```

Comment:
yml?

Reply:
yml and yaml are more or less interchangeable. But the official documentation suggests yaml so I will take it

Reply:
I heard of yml being a file extension (maybe to be compatible to DOS 8.3 file names) but never that yml is used to describe the file format.